src/main/scala/com/databricks/spark/avro/AvroSaver.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2015 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.databricks.spark.avro
+
+import java.sql.Timestamp
+import java.util.HashMap
+
+import scala.collection.immutable.Map
+
+import org.apache.spark.sql._
+import org.apache.spark.SparkContext._
+
+import org.apache.avro.generic.GenericData.Record
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.mapred._
+import org.apache.avro.SchemaBuilder.{RecordBuilder, FieldAssembler}
+import org.apache.avro.{SchemaBuilder, Schema}
+
+import org.apache.hadoop.io.NullWritable
+import org.apache.hadoop.mapred.JobConf
+
+
+/** 
+ * This object provides a save() method that is used to save SchemaRDD as avro file.
+ * To do this, we first convert the schema and then convert each row of the RDD to corresponding
+ * avro types. One remark worth mentioning is the structName parameter that functions have. Avro
+ * records have a name associated with them, which must be unique. Since SturctType in sparkSQL
+ * doesn't have a name associated with it, we are taking the name of the last structure field that
+ * the current structure is a child of. For example if the row at the top level had a field called
+ * "X", which happens to be a structure, we would call that structure "X". When we process original
+ * rows, they get a name "topLevelRecord".
+ */
+object AvroSaver {
+
+  private def convertTypeToAvro[T](dataType: DataType, 
+      schemaBuilder: SchemaBuilder.BaseTypeBuilder[T],
+      structName: String): T = {
+    dataType match {
+      case ByteType => schemaBuilder.bytesType()
+      case ShortType => schemaBuilder.intType()
+      case IntegerType => schemaBuilder.intType()
+      case LongType => schemaBuilder.longType()
+      case FloatType => schemaBuilder.floatType()
+      case DoubleType => schemaBuilder.doubleType()
+      case DecimalType() => schemaBuilder.stringType()
```

Comment:
I don't think this unapply exists in the newest version of Spark, and it would be unclear what it means if it did (since decimal may have a fixed precision). Let's just match using the type: `case _: DecimalType` here and everywhere else that uses DecimalType.